homeassistant/components/systemmonitor/sensor.py

```diff
@@ -70,185 +77,367 @@
 SIGNAL_SYSTEMMONITOR_UPDATE = "systemmonitor_update"
 
 
+def get_disk_use_percent(entity: SystemMonitorSensor[dataT]) -> float:
```

Comment:
In these functions we know what the generic type is. So we should replace the generic type with the actual type, I think. That should allow us to remove the use of cast below.